application/src/main/java/org/openhds/mobile/provider/DatabaseAdapter.java

```diff
@@ -255,6 +281,91 @@ public void updateCompleteStatus(long id, boolean completed) {
 		updateFormSubmission(id, cv);
 	}
 
+	//add recent forms paths
+	public long addHierarchyPath(String hierarchyPath, String filePath) {
```

Comment:
Naming is weird here too. We're not adding a hierarchy path, we're actually associatiing a form with the hierarchy path.